functions/schema-type-and-format.js

```diff
@@ -0,0 +1,76 @@
+// Check that format is valid for a schema type.
+// Valid formats are those defined in the OpenAPI spec and extensions in autorest.
+// - https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#data-types
+// - https://github.com/Azure/autorest/blob/main/packages/libs/openapi/src/v3/formats.ts
+
+// `input` is the schema of a request or response body
+module.exports = function checkTypeAndFormat(schema, options, { path }) {
+  if (schema === null || typeof schema !== 'object') {
+    return [];
+  }
+
+  const errors = [];
+
+  const stringFormats = [
+    // OAS-defined formats
+    'byte', 'binary', 'date', 'date-time', 'password',
+    // Additional formats recognized by autorest
+    'char', 'time', 'date-time-rfc1123', 'duration', 'uuid', 'base64url', 'url',
+    'odata-query', 'certificate',
```

Comment:
we want to support 'uri' as well, see https://github.com/Azure/azure-openapi-validator/issues/345